libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -161,6 +161,37 @@ def _should_bind_tools(model: LanguageModelLike, tools: Sequence[BaseTool]) -> b
     return False
 
 
+def _validate_messages(
+    messages: Sequence[BaseMessage],
+) -> None:
+    """Validate that all tool calls in AIMessages have a corresponding ToolMessage."""
+    all_tool_calls = [
+        tool_call
+        for message in messages
+        if isinstance(message, AIMessage)
+        for tool_call in message.tool_calls
+    ]
+    answered_tool_call_ids = {
+        message.tool_call_id for message in messages if isinstance(message, ToolMessage)
+    }
+    unanswered_tool_calls = [
+        tool_call
+        for tool_call in all_tool_calls
+        if tool_call["id"] not in answered_tool_call_ids
+    ]
+    if not unanswered_tool_calls:
+        return
+
+    error_message = (
+        "Found AIMessages with tool_calls that do not have a corresponding ToolMessage. "
+        f"Here are the first few of those tool calls: {unanswered_tool_calls[:3]}.\n\n"
+        "ALL tool calls in message history MUST have a corresponding ToolMessage - this is required by most LLM providers.\n\n"
```

Comment:
The **most** part is a problem in this error message. What if a user is working with an LLM that doesn't require it?

```suggestion
        "Every tool call (request to call a tool) in the message history MUST have a corresponding ToolMessage (result of tool invocation to be surfaced to the LLM) - this is required by most LLM providers.\n\n"
```

cc @baskaryan  -- should we enforce regardless?

Reply:
yea i don't love the imprecise language. i'd be in favor of enforcing regardless, but we could also do this validation only for specific LLMs too

Reply:
good point -- will double check this use case

Reply:
(although i suspect that might break react agent loop anyway)